src/components/shared/LessonSide/LevelSelector.tsx

```diff
@@ -1,5 +1,48 @@
-function LevelSelector(): JSX.Element {
-  return <div>Level Selector</div>;
+import '../../../styles/LessonSide.scss';
+
+interface LevelSelectorProps {
+  numLevels: number;
+  currentLevel: number;
+  maxLevelReached: number;
+}
+
+function LevelSelector({
+  numLevels,
+  currentLevel,
+  maxLevelReached,
+}: LevelSelectorProps): JSX.Element {
+  return (
+    <div className="selector-box">
+      <div className="level-selector">
+        {
+          //numLevels * 2 - 1 elements in Array as one button for each layer and numLevels - 1 connectors.
+          Array(numLevels * 2 - 1)
+            .fill(0)
+            .map((_, i) => {
+              const level = i / 2 + 1;
+              //If index is even, add a level button. Else, add a level connector.
+              return i % 2 == 0 ? (
+                <div
+                  className={
+                    'level-button' +
+                    (level === currentLevel ? ' current-level' : '')
+                  }
+                >
+                  {level}
+                </div>
+              ) : (
+                <div
+                  className={
+                    'level-connector' +
+                    (level < maxLevelReached ? ' reached-level-connector' : '')
+                  }
+                ></div>
```

Comment:
we can use visibility css so that we don't have to duplicate css! + also less work on client side
```suggestion
                <div
                  className={'level-connector'}
                  style={{"visibility": level < maxLevelReached ? "visible: "hidden"}}
                ></div>
```